demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,1271 @@
+            status = false;
+        }
+        else
+        {
+            long lenResult = -1;
+            /* Get the current position which is the file size. */
+            lenResult = ftell( file );
+
+            if( lenResult == -1 )
+            {
+                LogError( ( "Failed to get length of certificate signing request file. Path: %s. Error: %s.",
+                            PROVISIONING_CSR_PATH, strerror( errno ) ) );
+                status = false;
+            }
+            else
+            {
+                length = ( size_t ) lenResult;
+            }
+        }
+
+        if( status == true )
+        {
+            if( length > bufferLength )
+            {
+                LogError( ( "Buffer too small for certificate signing request. Buffer size: %ld. Required size: %ld.",
+                            bufferLength, length ) );
+                status = false;
+            }
+        }
+
+        if( status == true )
+        {
+            /* Return to the beginning of the file. */
+            result = fseek( file, 0L, SEEK_SET );
+
+            if( result == -1 )
+            {
+                LogError( ( "Failed to move to beginning of certificate signing request file. Path: %s. Error: %s.",
+                            PROVISIONING_CSR_PATH, strerror( errno ) ) );
+                status = false;
+            }
+        }
+
+        if( status == true )
+        {
+            size_t written = 0;
+            /* Read the CSR into our buffer. */
+            written = fread( buffer, 1, length, file );
+
+            if( written != length )
+            {
+                LogError( ( "Failed reading certificate signing request file. Path: %s. Error: %s.",
+                            PROVISIONING_CSR_PATH, strerror( errno ) ) );
+                status = false;
+            }
+            else
+            {
+                *outCsrLength = length;
+            }
+        }
+
+        fclose( file );
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool generateCsrRequest( char * buffer,
+                                size_t bufferLength,
+                                size_t * outLengthWritten )
+{
+    bool status = false;
+    char csr[ NETWORK_BUFFER_SIZE ] = { 0 };
+    size_t csrLength = 0;
+    CborEncoder encoder, mapEncoder;
+    CborError cborRet;
+
+    status = getCsr( csr, NETWORK_BUFFER_SIZE, &csrLength );
+
+    /* For details on the CreateCertificatefromCsr request payload format, see:
+     * https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html#create-cert-csr-request-payload
+     */
+    if( status == true )
+    {
+        cbor_encoder_init( &encoder, ( uint8_t * ) buffer, bufferLength, 0 );
+        cborRet = cbor_encoder_create_map( &encoder, &mapEncoder, 1 );
+
+        if( cborRet == CborNoError )
+        {
+            cborRet = cbor_encode_text_stringz( &mapEncoder, "certificateSigningRequest" );
```

Comment:
Error logging is missing for the return code of the CBOR functions

Reply:
Status is default false in the function. It is set to true at the end of the happy path. 
Additionally the return code is logged. If any CBOR error occurs, it will pass to the last else statement which logs the error.